chat-widget/src/components/livechatwidget/livechatwidgetstateful/LiveChatWidgetStateful.tsx

```diff
@@ -93,22 +93,48 @@ export const LiveChatWidgetStateful = (props: ILiveChatWidgetProps) => {
     const widgetElementId: string = props.controlProps?.id || "oc-lcw";
     const currentMessageCountRef = useRef<number>(0);
     let widgetStateEventName = "";
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let optionalParams: any;
+    let activeCachedChatExist = false;
+
+    if (!isUndefinedOrEmpty(state.appStates?.reconnectId)) {
+        activeCachedChatExist = true;
+        optionalParams = { reconnectId: state.appStates.reconnectId };
+    } else if (!isUndefinedOrEmpty(state.domainStates?.liveChatContext)) {
+        activeCachedChatExist = true;
+        optionalParams = { liveChatContext: state.domainStates?.liveChatContext };
+    } else {
+        activeCachedChatExist = false;
+        optionalParams = undefined;
+    }
 
-    const initiateEndChatOnBrowserUnload = () => {
-        TelemetryHelper.logActionEvent(LogLevel.INFO, {
-            Event: TelemetryEvent.BrowserUnloadEventStarted,
-            Description: "Browser unload event received."
-        });
-        endChat(props, chatSDK, setAdapter, setWebChatStyles, dispatch, adapter, false, false, false);
-        // Clean local storage
-        DataStoreManager.clientDataStore?.removeData(widgetStateEventName, "localStorage");
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const startChat = async (props: any, localState?: any) => {
+        let isChatValid = false;
+
+        //Start a chat from cache/reconnectid
+        if (activeCachedChatExist === true) {
+            dispatch({ type: LiveChatWidgetActionType.SET_CONVERSATION_STATE, payload: ConversationState.Loading });
+            if (localState) {
+                localState.appStates.conversationState = ConversationState.Loading;
+            }
 
-        //Dispose calling instance
-        if (voiceVideoCallingSDK) {
-            voiceVideoCallingSDK?.close();
+            //Check if conversation state is not in wrapup or closed state
+            isChatValid = await checkIfConversationStillValid(chatSDK, props, state.domainStates?.liveChatContext?.requestId);
+            if (isChatValid === true) {
+                await initStartChat(chatSDK, props.chatConfig, props.getAuthToken, dispatch, setAdapter, optionalParams);
+                return;
+            }
+        }
+
+        if (isChatValid === false) {
+            if (localState) {
+                await setPreChatAndInitiateChat(chatSDK, props.chatConfig, props.getAuthToken, dispatch, setAdapter, undefined, undefined, localState, props);
```

Comment:
if you are passing props already, why we need to pass props.chatConfig and getAuthToken separately?

Reply:
Like I mentioned above, I am not original author of this code. I fixed it for this instance, you would find same issues other places as well but that has to be handled in a separate PR by the dev.